client/src/app/components/audit-template/audit-template-view/create-audit-template/create-audit-template.component.spec.ts

```diff
@@ -5,36 +5,114 @@ import { RouterTestingModule } from '@angular/router/testing';
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
       declarations: [CreateAuditTemplateComponent],
       imports: [
         HttpClientTestingModule,
         RouterTestingModule,
-        MatAutocompleteModule
+        MatAutocompleteModule,
+        AppModule
       ],
       providers: [
         FormBuilder,
         {
           provide: AuditTemplateService,
         },
       ],
-    })
-      .compileComponents();
-  });
-
-  beforeEach(() => {
+    }).compileComponents();
+    
     fixture = TestBed.createComponent(CreateAuditTemplateComponent);
     component = fixture.componentInstance;
+    service = TestBed.inject(AuditTemplateService);
+
+    startDateInput = fixture.debugElement.query(By.css('#date')).nativeElement;
+    startTimeInput = fixture.debugElement.query(By.css('#time')).nativeElement;
+    templateInput = fixture.debugElement.query(By.css('#title')).nativeElement;
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+     // Test that Save button creates a new template
+  it('should create a new template', () => {
+      fixture.detectChanges();
+      spyOn(component.router, 'navigate');
+      const button = fixture.debugElement.query(By.css('#create-template-button')).nativeElement;
+      expect(button.disabled).toBeFalsy();
+  });
+
+  it('should be able to enter Start Date', () => {
+    fixture.detectChanges();
+    spyOn(component.router, 'navigate');
+    startDateInput.value = '2021-01-18T15:37:42';
+    startDateInput.dispatchEvent(new Event('change'));
+    expect(startDateInput.value).toBe('2021-01-18T15:37:42');
+});
+
+  it('should be able to enter Start Time', () => {
+    fixture.detectChanges();
+
+    startTimeInput.value = '23:33';
+    startTimeInput.dispatchEvent(new Event('change'));
+    expect(startTimeInput.value).toBe('23:33');
+});
+
+  it('should be able to enter Template name', () => {
+    fixture.detectChanges();
+    templateInput.value = 'Test 001';
+    templateInput.dispatchEvent(new Event('change'));
+    expect(templateInput.value).toBe('Test 001');
+});
+
+  // // Test the submitQuery()
+  // it('Call method', () => {
+  //   component.submitQuery({
+  //     location: '',
+  //     plant: '',
+  //     zones: '',
+  //     aisles: '',
+  //     bins: '',
+  //     part_number: '',
+  //     serial_number: '',
+  //     start_date: '',
+  //     repeat_every: '',
+  //     on_day: '',
+  //     for_month: '',
+  //     time_zone_utc: '',
+  //     });
+  //     expect(component.errorMessage).toBe('Please choose at least one day');
+  //   });
+
+  // it('should be able to submit a new query', () => {
+  //   try {
+  //     component.submitQuery([{
+  //       dayArray: [],
+  //       monthArray: [],
+  //       year: 2020,
+  //       month: ,
+  //       day: ,
+  //       hour: ,
+  //       minute: ,
+  //     }])
+  //   }
+  //   catch (error) {
+  //   console.error(error);
+  //   }
+  // });
```

Comment:
if this isn't needed, can we delete it?

Reply:
As Avery mentioned above, this is commented out is to show others how to setup their tests / serve as an example.